packages/react/src/evaluation/use-feature-flag.ts

```diff
@@ -298,17 +297,30 @@ function attachHandlersAndResolve<T extends FlagValue>(
     suspendUntilReconciled(client);
   }
 
-  const [evaluationDetails, setEvaluationDetails] = useState<EvaluationDetails<T>>(
+  const [evaluationDetails, setEvaluationDetails] = useState<EvaluationDetails<T>>(() =>
     resolver(client).call(client, flagKey, defaultValue, options),
   );
+  
+  // Re-evaluate when dependencies change (handles prop changes like flagKey)
```

Comment:
I think this comment really should be enhanced to describe the other reason for this `useEffect` (and the primary focus of this PR); to re-render if the evaluation details has changed (this is VERY different from a prop changing, which is all the comment says).

Reply:
https://github.com/open-feature/js-sdk/pull/1226/files#r2279726954